utils/git_diff.py

```diff
@@ -0,0 +1,371 @@
+    "modeling_flax_utils.py": "test_modeling_flax_common.py",
+    "modeling_tf_utils.py": "test_modeling_tf_common.py",
+    "modeling_utils.py": ["test_modeling_common.py", "test_offline.py"],
+    "models/auto/modeling_auto.py": ["test_modeling_auto.py", "test_modeling_tf_pytorch.py", "test_modeling_bort.py"],
+    "models/auto/modeling_flax_auto.py": "test_flax_auto.py",
+    "models/auto/modeling_tf_auto.py": [
+        "test_modeling_tf_auto.py",
+        "test_modeling_tf_pytorch.py",
+        "test_modeling_tf_bort.py",
+    ],
+    "models/blenderbot_small/tokenization_blenderbot_small.py": "test_tokenization_small_blenderbot.py",
+    "models/blenderbot_small/tokenization_blenderbot_small_fast.py": "test_tokenization_small_blenderbot.py",
+    "models/gpt2/modeling_gpt2.py": ["test_modeling_gpt2.py", "test_modeling_megatron_gpt2.py"],
+    "pipelines/base.py": "test_pipelines_common.py",
+    "pipelines/text2text_generation.py": [
+        "test_pipelines_text2text_generation.py",
+        "test_pipelines_summarization.py",
+        "test_pipelines_translation.py",
+    ],
+    "pipelines/zero_shot_classification.py": "test_pipelines_zero_shot.py",
+    "testing_utils.py": "test_skip_decorators.py",
+    "tokenization_utils.py": "test_tokenization_common.py",
+    "tokenization_utils_base.py": "test_tokenization_common.py",
+    "tokenization_utils_fast.py": "test_tokenization_fast.py",
+    "trainer.py": [
+        "test_trainer.py",
+        "extended/test_trainer_ext.py",
+        "test_trainer_distributed.py",
+        "test_trainer_tpu.py",
+    ],
+    "utils/versions.py": "test_versions_utils.py",
+}
+
+
+def module_to_test_file(module_fname):
+    """
+    Returns the name of the file(s) where `module_fname` is tested.
+    """
+    splits = module_fname.split(os.path.sep)
+
+    # Special map has priority
+    short_name = os.path.sep.join(splits[2:])
+    if short_name in SPECIAL_MODULE_TO_TEST_MAP:
+        test_file = SPECIAL_MODULE_TO_TEST_MAP[short_name]
+        if isinstance(test_file, str):
+            return f"tests/{test_file}"
+        return [f"tests/{f}" for f in test_file]
+
+    module_name = splits[-1]
+    # Fast tokenizers are tested in the same file as the slow ones.
+    if module_name.endswith("_fast.py"):
+        module_name = module_name.replace("_fast.py", ".py")
+
+    # Special case for pipelines submodules
+    if len(splits) >= 2 and splits[-2] == "pipelines":
+        default_test_file = f"tests/test_pipelines_{module_name}"
+    # Special case for benchmarks submodules
+    elif len(splits) >= 2 and splits[-2] == "benchmark":
+        return ["tests/test_benchmark.py", "tests/test_benchmark_tf.py"]
+    # Special case for commands submodules
+    elif len(splits) >= 2 and splits[-2] == "commands":
+        return "tests/test_cli.py"
+    # Special case for onnx submodules
+    elif len(splits) >= 2 and splits[-2] == "onnx":
+        return ["tests/test_onnx.py", "tests/test_onnx_v2.py"]
```

Comment:
I think there's, unfortunately, something that we'll need to update either here or in our manner of testing things. The ONNX test are a very good example of this, but I doubt it's the only one.

Here, if modifying any file in the `onnx` directory, the `test_onnx*.py` tests will run. Great! However, that's not the only tests that a modification to the ONNX files should trigger: there is also the following test, in `test_modeling_tf_common.py`:

https://github.com/huggingface/transformers/blob/662a556528dcf9eaf6fd91e060b7b9dc34bbe765/tests/test_modeling_tf_common.py#L328-L345

There are a few others in that file. Unfortunately, the tests aren't self-contained as these need to be run on all models that have ONNX export capabilities.

The questions this raises is:
- Are there a lot of similar cases
- If so, how do we handle them?

If the answer is "no" to the first question, then that's not too much of an issue.